src/Events/Services/SubscriptionService.cs

```diff
@@ -8,150 +9,156 @@
 using Altinn.Platform.Events.Models;
 using Altinn.Platform.Events.Repository;
 using Altinn.Platform.Events.Services.Interfaces;
-using Altinn.Platorm.Events.Extensions;
 
-namespace Altinn.Platform.Events.Services
+namespace Altinn.Platform.Events.Services;
+
+/// <inheritdoc/>
+public class SubscriptionService : ISubscriptionService
 {
-    /// <inheritdoc/>
-    public class SubscriptionService : ISubscriptionService
+    private readonly ISubscriptionRepository _repository;
+    private readonly IEventsQueueClient _queue;
+    private readonly IClaimsPrincipalProvider _claimsPrincipalProvider;
+    private readonly IAuthorization _authorization;
+
+    private const string _organisationPrefix = "/organisation/";
+    private const string _orgPrefix = "/org/";
+    private const string _systemUserPrefix = "/systemuser/";
+    private const string _userPrefix = "/user/";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SubscriptionService"/> class.
+    /// </summary>
+    public SubscriptionService(
+        ISubscriptionRepository repository,
+        IAuthorization authorization,
+        IEventsQueueClient queue,
+        IClaimsPrincipalProvider claimsPrincipalProvider)
+    {
+        _repository = repository;
+        _authorization = authorization;
+        _queue = queue;
+        _claimsPrincipalProvider = claimsPrincipalProvider;
+    }
+
+    /// <summary>
+    /// Completes the common tasks related to creating a subscription once the producer specific services are completed
+    /// </summary>
+    internal async Task<(Subscription Subscription, ServiceError Error)> CompleteSubscriptionCreation(Subscription eventsSubscription)
     {
-        private readonly ISubscriptionRepository _repository;
-        private readonly IEventsQueueClient _queue;
-        private readonly IClaimsPrincipalProvider _claimsPrincipalProvider;
-        private readonly IAuthorization _authorization;
-
-        private const string OrgPrefix = "/org/";
-        private const string UserPrefix = "/user/";
-        private const string OrganisationPrefix = "/organisation/";
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="SubscriptionService"/> class.
-        /// </summary>
-        public SubscriptionService(
-            ISubscriptionRepository repository,
-            IAuthorization authorization,
-            IEventsQueueClient queue,
-            IClaimsPrincipalProvider claimsPrincipalProvider)
+        if (!await _authorization.AuthorizeConsumerForEventsSubscription(eventsSubscription))
```

Comment:
I was wrong when we talked about this. We do need to update the `XacmlMapperHelper.AddSubjectAttribute()` method. It's used in a call chain (here) where subject is the event subscriber. I thought it was limited to when subject is the "instance owner" of an instance.

The term Subject is used in two different contexts.
1. The active user trying to access some data/event.
2. The subject of an event.